dataplane/dataplane.cpp

```diff
@@ -1327,6 +1348,28 @@ eResult cDataPlane::splitSharedMemoryPerWorkers()
 
 			ring_id++;
 		}
+		// init lowPriority ring
+		{
+			auto name = "r_lp_" + std::to_string(core_id);
+			auto offset = offsets[shm];
+			auto memaddr = (void*)((intptr_t)shm + offset);
+			sharedmemory::cSharedMemory ring;
+
+			auto unit_size = sizeof(sharedmemory::item_header_t) + MAX_PACK_SIZE;
+			if (unit_size % RTE_CACHE_LINE_SIZE != 0)
+			{
+				unit_size += RTE_CACHE_LINE_SIZE - unit_size % RTE_CACHE_LINE_SIZE; /// round up
+			}
+			const auto units_number = getConfigValues().ring_lowPriority_size;
+			const auto size = sizeof(sharedmemory::ring_header_t) + unit_size * units_number;
+			ring.init(memaddr, unit_size, units_number);
+			offsets[shm] += size;
+
+			worker->lowPriorityRing = ring;
+
+			auto meta = common::idp::get_shm_info::dump_meta(name, "lp", unit_size, units_number, core_id, socket_id, key, offset);
+			dumps_meta.emplace_back(meta);
+		}
```

Comment:
This almost looks like a previous block of code (body of `for (const auto& [tag, ring_cfg] : config.shared_memory)` loop).
I think we could extract common functionality to a function.
something like:
```cpp
  void foo(arguments)
  {
      auto name = base_name + std::to_string(core_id) + (is_low_priority ? "" : "_" + std::to_string(ring_id));
  
      auto unit_size = sizeof(sharedmemory::item_header_t) + dump_size;
      if (unit_size % RTE_CACHE_LINE_SIZE != 0)
      {
          unit_size += RTE_CACHE_LINE_SIZE - unit_size % RTE_CACHE_LINE_SIZE;
      }
  
      auto size = sizeof(sharedmemory::ring_header_t) + unit_size * units_number;
      auto offset = offsets[shm];
      auto memaddr = reinterpret_cast<void*>(reinterpret_cast<std::uintptr_t>(shm) + offset);
  
      sharedmemory::cSharedMemory ring;
      ring.init(memaddr, unit_size, units_number);
      offsets[shm] += size;
  
      auto meta = common::idp::get_shm_info::dump_meta(name, tag, unit_size, units_number, core_id, socket_id, key, offset);
      dumps_meta.emplace_back(meta);
  }
```
as we cannot interact with private `worker` fields in the regular function.

Then the "low priority ring" turns into:
```
foo(...);
worker->lowPriorityRing = ring;
```

and loop body turns into
```
foo();
worker->dumpRings[ring_id] = ring;                                                                                                                                  
tag_to_id[tag] = ring_id;                                          
ring_id++;  
```